edb/schema/indexes.py

```diff
@@ -91,14 +91,39 @@ def is_index_valid_for_type(
         case 'fts::index':
             return is_subclass_or_tuple(expr_type, 'fts::document', schema)
         case 'pg::gist':
-            return expr_type.is_range() or expr_type.is_multirange()
+            return (
+                expr_type.is_range()
+                or
+                expr_type.is_multirange()
+                or
+                expr_type.issubclass(
+                    schema,
+                    (
+                        schema.get('ext::postgis::geometry',
```

Comment:
Could you add a `TODO` comment here reminding us to remove these *and* reminding us to remove the `Optional` hack in `issubclass` when we do?

Reply:
Added